tests/ga/test_extension.py

```diff
@@ -2535,304 +2536,285 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
 
-    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(
-            self, *args):
-        old_handler_i = self._get_ext_handler_instance('foo', '1.0.0')
-        new_handler_i = self._get_ext_handler_instance('foo', '1.0.1', continue_on_update_failure=True)
+    def test_non_enabled_ext_should_not_be_disabled_at_ver_update(self):
 
-        with patch.object(CGroupConfigurator.get_instance(), "start_extension_command",
-                          side_effect=['ok', 'ok', ExtensionError('uninstall Failed'), 'ok']) as patch_start_cmd:
+        first_ext = get_extension_actor(enableAction=Actions.FailAction)
+        second_ext = get_extension_actor(version="1.1.0")
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        second_ext.get_command("installCommand").assert_called_once()
+        second_ext.get_command("enableCommand").assert_called_once()
+        first_ext.get_command("disableCommand").assert_not_called()
+        
 
-            ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+    def test_disable_failed_env_variable_should_be_set_for_update_cmd_when_continue_on_update_failure_is_true(self):
+        exit_code = uuid.uuid4()    # Generate a unique UUID for highest strictness.
+        unique_fail = lambda *args, **kwargs: exit_code
 
-            args, kwargs = patch_start_cmd.call_args
+        first_ext = get_extension_actor(disableAction=unique_fail)
+        second_ext = get_extension_actor(version="1.1.0", continueOnUpdateFailure=True)
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        first_ext.get_command("disableCommand").assert_called_once()            
+        second_ext.get_command("updateCommand").assert_called_once()
+        first_ext.get_command("uninstallCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
+        
+        _, kwargs = second_ext.get_command("updateCommand").call_args
+
+        assert kwargs.get("env", {}).get(ExtCommandEnvVariable.DisableReturnCode, "") == str(exit_code)
+
+
+    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(self):
+
+        exit_code = uuid.uuid4()    # Generate a unique UUID for highest strictness.
+        unique_fail = lambda *args, **kwargs: exit_code
+
+        first_ext = get_extension_actor(uninstallAction=unique_fail)
+        second_ext = get_extension_actor(version="1.1.0", continueOnUpdateFailure=True)
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        first_ext.get_command("disableCommand").assert_called_once()            
+        second_ext.get_command("updateCommand").assert_called_once()
+        first_ext.get_command("uninstallCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
```

Comment:
How are we validating that the order of these calls is correct?

Reply:
We are not validating ordering as of now. In another thread, I've mentioned that I can get that working, but wasn't sure if that was something we wanted to be part of the ExtensionActor class.

Reply:
We need to validate the order of commands, not just that they were called.

Reply:
I agree, and I believe @narrieta does as well. As I see it, there are two options:

1. Do it manually in every single test.
2. Bake it into the `extension_actor` module as a function/method.

My idea for option 1 is a list that can be appended to from within lambdas local to the test; the keys added to the list can then be checked for the correct order after the fact.

It seems like this is something we'd want to do on basically every test dealing with extensions, though, so it would be great to extend that idea into something that works with option 2. I'm running into some scoping problems, however. I think I need to think about this some more. If you have an alternative, let me know.